kategory-docs/docs/docs/datatypes/state/README.md

```diff
@@ -6,4 +6,95 @@ permalink: /docs/datatypes/state/
 
 ## State 
 
-TODO
\ No newline at end of file
+`State` is a structure that provides a functional approach to handling application state. `State<S, A>` is basically a function `S -> Tuple2(S, A)`, where `S` is the type that represents your state and `A` is the result the function produces. In addition to returning the result of type `A`, the function returns a new `S` value, which is the updated state.
+
+## Stack
+
+Let's build a simple Stack using an immutable data structure like Kotlin's `List`:
+
+```kotlin:ank:silent
+typealias Stack = List<String>
```

Comment:
Let's use `NonEmptyList` here so there is no need to throw an `IllegalStateException` when contemplating an `emptyList`. Either that or make `pop` return a `Option<Pair<Stack, String>>`